engine/runtime/src/memories/noncontiguous_memory.rs

```diff
@@ -90,14 +90,14 @@ impl LockedMemory for NonContiguousMemory {
     /// Unlocks the memory and returns an unlocked Buffer
     /// To retrieve secret value you xor the hash contained in shard1 with value in shard2
     fn unlock(&self) -> Result<Buffer<u8>, MemoryError> {
-        let (data1, data2) = self.get_shards_data()?;
-        let data1 = &blake2b::Blake2b256::digest(data1);
-        let reconstructed_data = xor(data1, &data2, NC_DATA_SIZE);
+        let (r, mut m) = self.get_shards_data()?;
+        let h = &blake2b::Blake2b256::digest(r);
+        xor_mut(&mut m, h, NC_DATA_SIZE);
```

Comment:
I guess it's fine but just to be sure, the params are not in the same order as before, is that fine?

Reply:
r - randomness; hr=h(r) - hash of randomness; m - secret message.
shards = (r, m^hr)
To open message from shards: m = shard_2 ^ h(shard_1) = m^hr ^ h(r) = m^hr^hr

Reply:
It's almost like these variables should be named appropriately :trollface: